server/server/src/serve_frontend.rs

```diff
@@ -10,13 +13,33 @@ use rust_embed::RustEmbed;
 #[folder = "../../client/packages/host/dist"]
 struct Asset;
 
-const INDEX: &str = "index.html";
+const INDEX: &'static str = "index.html";
+const CACHE_MAX_AGE: u32 = 365 * 60 * 60 * 24; // 1 year
 
 // https://github.com/pyrossh/rust-embed/blob/master/examples/actix.rs
 fn serve_frontend(path: &str) -> HttpResponse {
     if let Some(content) = Asset::get(path) {
+        let cache_control = if path == "index.html" {
+            // The index and config files can change so we don't want to cache them
+            // The other files are generally static and can be cached
+            // Technically the config.js shouldn't change either but if it did we'd want pick it up immediately.
+            header::CacheControl(vec![header::CacheDirective::NoCache])
+        } else if path.starts_with("locales/") {
+            // These are the translation json files, in the typescript code they are cached in local storage and invalidated after a yarn build
+            // So we don't want to cache them here...
+            header::CacheControl(vec![header::CacheDirective::NoCache])
```

Comment:
In what scenario are they changing? Aren't all frontend files bundled into the server binary and don't change at all? And when doing dev we run the frontend separately, and they are not coming from the server?

Reply:
In the old code the files are cached by the user's browser, so if you're connecting via the web browser to the server, and there's a server upgrade, the browser won't re-fetch the translation files. More of a problem for Health Supply Hub that for OMS as generally we're using android or a electron but doesn't hurt to fix :)